packages/visitor-keys/src/visitor-keys.ts

```diff
@@ -258,7 +253,9 @@ const additionalKeys: AdditionalKeys = {
   TSTupleType: ['elementTypes'],
   TSTypeAliasDeclaration: ['id', 'typeParameters', 'typeAnnotation'],
   TSTypeAnnotation: ['typeAnnotation'],
-  TSTypeAssertion: SharedVisitorKeys.TypeAssertion,
```

Comment:
Also noting for the record that this was previously, apparently unintentionally, flipped in https://github.com/typescript-eslint/typescript-eslint/pull/5926/files#diff-04c8c6315c623a82f3004fbfe093162ab6811b1f207cf1832d495b2979466f5eL139